Objetivo 6

XileonXL opened the pull request:
* [x] ¿Se ha evitado Circle CI, porque a estas alturas ya lo ha usado todo el mundo?
* [x] ¿Se han establecido criterios *a priori* para elegir el sistema de integración continua más conveniente y se han calificado los sistemas que cumplan esos requisitos según esos criterios?
* [x] ¿Se han examinado varios sistemas de integración continua? 
* [x] ¿Se han configurado varios sistemas de integración continua? 
* [x] ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar desde los tests)? 
* [x] ¿Uno de los sistemas configurados permite comprobar cuales son las versiones dellenguaje con las que funciona correctamente nuestra aplicación? 
* [x] ¿Se escogen de forma adecuada las versiones del lenguaje que se testean, tanto en el sistema de CI como en el contenedor Docker? 
* [x] ¿Se han justificado correctamente las versiones del lenguaje que se están testeando y se ha comprobado que no se comprueban varias veces lo mismo (la misma versión en CI y en el contenedor Docker en otro sistema CI?

Todas las respuestas a estas preguntas se encuentran implícitamente o explícitamente [aquí](https://github.com/Xileon310/GoParty/pull/44/files#diff-80e64b7137dd7009efdbd313b65a1e1092a49bb565596a031194cb73f723bafe)

XileonXL committed:
Añade clave CI. Avanza #41

XileonXL committed:
Añade fichero configuración CircleCi. Avanza #41

Se añade configuración para obtener imagen de DockerHub y ejecución de
tests haciendo uso de Task.

XileonXL committed:
Añade configuración para Buddy. Avanza #42

Se añade configuración para el sistema de integración Buddy.
Se ejecutará una versión de Go.

XileonXL committed:
Corrige errata en documentación

XileonXL committed:
Añade ruta a doc. Avanza #40 #41 #42

XileonXL committed:
Cambia imagen docker

XileonXL committed:
Añade información de CIs. Closes #40

Se añade comparativa entre CIs.
Se añade información de las tareas y versiones de golang utilizadas.
Se añade conclusión y elección de CIs.

XileonXL committed:
Corrige errata extension. Closes #42

XileonXL committed:
Corrige errata. Closes #41

XileonXL committed:
Corrige errata en documentación

XileonXL committed:
Elimina branches del workflow de imagen docker

Xileon310 commented:
Estaba desarrollando el objetivo en base a los criterios anteriores, y puesto que en ordinaria lo entregué con Circle CI y Github Actions, no me había percatado del primer requisito :sob: 

Por lo que cambiaré la documentación y configuración para un nuevo CI en los siguientes commits.

XileonXL committed:
Añade configuración de Gitlab. #41

XileonXL committed:
Pruebo GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Configuro GitLab. #41

XileonXL committed:
Probando Travis CI. #41

XileonXL committed:
Elimino Travis CI. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Pruebo Github Actions. #41

XileonXL committed:
Elimino Git Lab. #41

XileonXL committed:
Añade documentación final de los CIs. Closes #41

Se cambia Circle CI por Github Actions.
Se argumenta el cambio.
Se añaden conclusiones y requisito del objetivo.

Xileon310 commented:
## Objetivo 6

Se han realizado los siguientes cambios:
- Se añade nueva clave a iv.yaml.
- Se añade documentación referente a los CIs, respaldada por los issues #40 #41 y #42 que se basan en la HU #27 .
- Se añade fichero de configuración para CircleCI y Buddy.
- Se añade workflow para Github Action.

XileonXL committed:
Añade información de Git Lab. Closes #40

XileonXL committed:
Corrige error de ejecución. #41

XileonXL committed:
Cambia workflow de Buddy. #42

Buddy ahora comprobará la imagen de Docker y ejecutará tests desde task
runner.

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Cambia workflow de Github Actions. #41

Ahora el workflow comprobará la versión 1.16.13 de golang.

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Elimima pull_request option

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Elimino Circleci

XileonXL committed:
Añado otra vez Circle CI

XileonXL committed:
Corrige error en Buddy. #42

XileonXL committed:
Elimino Buddy

XileonXL committed:
Añado Buddy para ejecutar golang en 1.16.13. #41

Xileon310 commented:
**Puesto que ya había desarrollado el objetivo para Circle CI, he dejado el archivo de configuración correspondiente, aunque no cuente para nota. Se explica todo en el fichero de documentación correspondiente a los CIs.**

También comento mi imposibilidad para hacer funcionar Git Lab. Nunca más volveré a usarlo.

Por otro lado, he intentado hacer pull del contenedor en Git Lab y Buddy, exactamente haciendo lo mismo que con Circle CI. En ambos me daba problemas del task runner, mientras que en Circle CI funciona perfectamente. Lo siento, he estado horas intentándolo y he fracasado. Cualquier ayuda podría ser útil.

## Objetivo 6

Se han realizado los siguientes cambios:
- Se añade nueva clave a iv.yaml.
- Se añade documentación referente a los CIs, respaldada por los issues #40 #41 y #42 que se basan en la HU #27 .
- Se añade fichero de configuración para CircleCI y Buddy.
- Se añade workflow para Github Action.

JJ commented:
⛹ Revisores → @amerigal @Pablont98 @modejota

XileonXL committed:
Vuelvo a probar Gitlab. #41

XileonXL committed:
Vuelvo a probar Gitlab. #41

XileonXL committed:
Vuelvo a probar Gitlab. #41

XileonXL committed:
Añade acción de Buddy. #42

XileonXL committed:
Elimina acción de Buddy. #42

XileonXL committed:
Elimino Gitlab. #41

XileonXL committed:
Pruebo Azure CI. #41

XileonXL committed:
Elimino Azure. #41

XileonXL committed:
Añade acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Pruebo acción de Buddy. #42

XileonXL committed:
Elimino acción de Buddy. #42

XileonXL committed:
Añado acción de Buddy. #42

XileonXL committed:
Elimino acción de Buddy. #42

XileonXL committed:
Pruebo GitLab. #41

XileonXL committed:
Pruebo GitLab. #41

XileonXL committed:
Elimino Gitlab. #41

XileonXL committed:
Amplio documentación con problemas que han ocurrido. #40

Xileon310 commented:
**Puesto que ya había desarrollado el objetivo para Circle CI, he dejado el archivo de configuración correspondiente, aunque no cuente para nota. Se explica todo en el fichero de documentación correspondiente a los CIs.**

Por otro lado, he intentado hacer pull del contenedor en Git Lab y Buddy, exactamente haciendo lo mismo que con Circle CI. En ambos me daba problemas del task runner, mientras que en Circle CI funciona perfectamente. El problema reside en que en ambos CI, el task runner no detecta el Taskfile en la ruta en la que se debería cargar, pero con Circle CI no hay ningún problema (literalmente es el mismo archivo cambiando claves y estructura diferente para los 3). Lo siento, he estado horas intentándolo y he fracasado. Cualquier ayuda podría ser útil, puedo aportar los .yml utilizados para cada CI e incluso una guía de cómo lo he intentado hacer.

Además de los anteriores CI, he intentado hacerlo con Azure CI, pero una vez desplegado, configurado y registrado la cuenta, me ha pedido tarjeta de crédito para permitir usar los pipelines con minutos gratuitos. Una vez añadida la tarjeta de crédito y tras 30 minutos peleandome con la maravillosa suite de Microsoft, me ha notificado que no soy elegible para tal oferta, por lo que he eliminado dicho archivo.

## Objetivo 6

Se han realizado los siguientes cambios:
- Se añade nueva clave a iv.yaml.
- Se añade documentación referente a los CIs, respaldada por los issues #40 #41 y #42 que se basan en la HU #27 .
- Se añade fichero de configuración para CircleCI y Buddy.
- Se añade workflow para Github Action.

XileonXL committed:
Corrijo estructura de tabla en doc. #40

Xileon310 commented:
@JJ Listo para revisión.

JJ commented on .github/workflows/test-image.yml:
```diff
@@ -0,0 +1,17 @@
+          go-version: 1.16.13
+
+      - name: Run tests
+        run: go test -v ./...
```
Los tests siempre tienes que ejecutarlos desde el task runner.

JJ reviewed (approved):
:+1: porque el objetivo está cumplido, aunque te agradecería que usaras realmente el task runner que has configurado desde los sistemas de CI, que para eso está.
